File: src/com/pwootage/fasterflux/config/FasterFluxConfig.java
package com.pwootage.fasterflux.config;

import java.io.File;

import net.minecraftforge.common.Configuration;

public class FasterFluxConfig  {
	public static final String CATEGORY_OTHER = "other";
	private File confDir;
	private Configuration conf;
	
	public final int multiblockId1;
	public final int syncDelay;
	
	public FasterFluxConfig(File file) {
		confDir = file;
		conf = new Configuration(new File(confDir, "fasterflux.cfg"));
		
		conf.load();
		
		//Blocks
		multiblockId1 = conf.getBlock("multiblockId1", 950, "Main multiblock ID").getInt();
		
		//Other configs
		syncDelay = conf.get(CATEGORY_OTHER, "syncDelay", 1, "Delay between sync packets (ticks, can reduce network usage)").getInt();
		
		//Done!
		conf.save();
	}
}

File: src/com/pwootage/fasterflux/config/FasterFluxConfig.scala
package com.pwootage.fasterflux.config

import java.io.File
import net.minecraftforge.common.Configuration
import FasterFluxConfig._

object FasterFluxConfig {

  val CATEGORY_OTHER = "other"
}

class FasterFluxConfig(private var confDir: File) {

  private var conf: Configuration = new Configuration(new File(confDir, "fasterflux.cfg"))

  conf.load()
  
  val multiblockId1 = conf.getBlock("multiblockId1", 950, "Main multiblock ID").getInt

  val syncDelay = conf.get(CATEGORY_OTHER, "syncDelay", 1, "Delay between sync packets (ticks, can reduce network usage)").getInt

  conf.save()
}
